libs/wingsdk/src/target-tf-aws/api.ts

```diff
@@ -0,0 +1,157 @@
+/* eslint-disable import/order */
+import { join } from "path";
+import { ApiGatewayMethod } from "@cdktf/provider-aws/lib/api-gateway-method";
+import { ApiGatewayResource } from "@cdktf/provider-aws/lib/api-gateway-resource";
+import { ApiGatewayRestApi } from "@cdktf/provider-aws/lib/api-gateway-rest-api";
+import { ApiGatewayDeployment } from "@cdktf/provider-aws/lib/api-gateway-deployment";
+import { ApiGatewayStage } from "@cdktf/provider-aws/lib/api-gateway-stage";
+import { ApiGatewayMethodSettings } from "@cdktf/provider-aws/lib/api-gateway-method-settings";
+import { LambdaPermission } from "@cdktf/provider-aws/lib/lambda-permission";
+import { DataAwsRegion } from "@cdktf/provider-aws/lib/data-aws-region";
+import { DataAwsCallerIdentity } from "@cdktf/provider-aws/lib/data-aws-caller-identity";
+
+import { Construct } from "constructs";
+import { core } from "..";
+import * as cloud from "../cloud";
+import { convertBetweenHandlers } from "../convert";
+import { Code } from "../core/inflight";
+import { Function } from "./function";
+import { ApiGatewayIntegration } from "@cdktf/provider-aws/lib/api-gateway-integration";
+import { IResource } from "../core";
+
+/**
+ * AWS Implementation of `cloud.Api`.
+ */
+export class Api extends cloud.ApiBase {
+  private readonly api: ApiGatewayRestApi;
+  constructor(scope: Construct, id: string, props: cloud.ApiProps = {}) {
+    super(scope, id, props);
+    this.api = new ApiGatewayRestApi(this, "api", {
+      name: id,
+    });
+  }
+
+  public get(
+    route: string,
+    inflight: core.Inflight,
+    props?: cloud.ApiOnGetProps
+  ): void {
+    const hash = inflight.node.addr.slice(-8);
+
+    const apiResource = new ApiGatewayResource(this, `resource-${hash}`, {
+      restApiId: this.api.id,
+      parentId: this.api.rootResourceId,
+      pathPart: route,
+    });
+
+    const apiMethod = new ApiGatewayMethod(this, `method-${hash}`, {
+      restApiId: this.api.id,
+      resourceId: apiResource.id,
+      httpMethod: "GET",
+      // FIXME: this cannot be hardcoded
+      authorization: "NONE",
+    });
+
+    const functionHandler = convertBetweenHandlers(
+      this.node.scope!, // ok since we're not a tree root
```

Comment:
For Api it's okay if we make all each of the route resources be children of `Api` (instead of siblings)
```suggestion
      this,
```

Reply:
ok, I'll change that. Can you explain me what that function does? Is this convert actually needed in this case?

Reply:
Sure, I'd be happy to. The purpose of `convertBetweenHandlers` is to perform a conversion to and from the APIs that the Wing user implements against in their Wing code, and the interface of the input/output expected by the cloud provider.

For example, in Wing, the end goal is to let a user write something like:

```js
bring cloud;

let api = new cloud.Api();
api.get("/ducks", inflight (req: cloud.ApiRequest): ApiResponse => {
  // do stuff with req
  return ApiResponse {
    status: 200,
    body: "Ducks created"
  };
});
```

When this is implemented on other clouds like Azure and GCP, it will still be the same code! However, an AWS Lambda that is hooked up to an AWS API gateway receives a different shape of input, and expect a different shape of output. Perhaps it expects `statusCode` instead of `status`, or includes an `httpMethod` instead of `method`. That's fine, as long as the SDK wraps the user's inflight code in another function that transforms the fields for it.

So, conceptually that's the purpose. In terms of implementation, the function accepts an `Inflight` (class from the SDK) which is treated as an immutable data structure. Since it's immutable, we cannot mutate it with the wrapped code, and instead it creates a new `Inflight` with the new code. (At least, that's how it works at the moment). Let me know if that explanation helps at all 😅